### PR TITLE
[Extends #3738] update plugin docs and package gitresolver + ruby plugin

### PR DESCRIPTION
**Pull request by robaerd**
Extends #3738

This PR adds/modifies following:
- [x] reformat install.md instructions
- [x] document plugin/binding package names
- [x] add general statement for plugin/binding installation
- [x] package gitresolver plugin, ruby plugin and ruby bindings
- [x] add instructions to run spellchecker locally #3760

## Basics

These points need to be fulfilled for every PR:

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by robaerd** [comment]
@markus2330 should I also add such a line to the plugins that are part of the `libelektra5` package, or should we just state in the src/plugins/README.md something along the lines of:
```
Many plugins are part of the core package `libelektra5`. The package that includes a plugin which does not belong to the `libelektra5` package can be found in it's README.md.
```


I also noticed that the `yanlr` plugin does not belong to any package (defaults to elektra-misc). It currently gets not packaged at all, as the anltr4 dependency is never fulfilled. Maybe we should still include it into the libelektra5-experimental package?

Also, what exactly is the reason that we do not package the `gitresolver` and `ruby` plugin?

**Comment by markus2330** [comment]
> @markus2330 should I also add such a line to the plugins that are part of the libelektra5 package

If they are present in the core package probably no specific information about installation is needed. Having one general statement is a good idea, though.

> I also noticed that the yanlr plugin does not belong to any package (defaults to elektra-misc). It currently gets not packaged at all, as the anltr4 dependency is never fulfilled. Maybe we should still include it into the libelektra5-experimental package?

I don't know, I think the yamlcpp is the recommended one and yanlr is not really needed. It is more a technology study. @sanssecours What do you think? Do you want to maintain it? Should we package it? Or do you want to remove the plugin?

> Also, what exactly is the reason that we do not package the gitresolver and ruby plugin?

gitresolver was always a bit problematic as it needed a quite recent version of libgit. As this is probably not a problem anymore, we can actually package it.

The ruby plugin probably was simply forgotten.

It would be nice to have these two in (separate) packages. It is not very urgent though, as they are quite experimental.

**Comment by sanssecours** [comment]
> @sanssecours I don't know, I think the yamlcpp is the recommended one and yanlr is not really needed. It is more a technology study. @sanssecours What do you think? Do you want to maintain it? 

I do not think I have time to maintain the plugin. If nobody else is interested, then it might be best to just remove it.

> Should we package it? Or do you want to remove the plugin?

I deleted the Yan LR and YAML Smith plugins [here](https://github.com/ElektraInitiative/libelektra/pull/3752).

**Comment by mpranj** [comment]
Thank you! Can you please rebase?

File: doc/news/_preparation_next_release.md [hunk 1]
```diff
@@ -148,6 +148,21 @@ you up to date with the multi-language support provided by Elektra.
...
+- Reworked [METADATA.ini](/doc/METADATA.ini) _(Markus Raab)_
+- Minor rewording in INSTALL.md _(@a-kraschitzer)_
+- Write notes that `\\` are due to shell recorder, and are not to be copied _(Markus Raab)_
+- Add link to [Go](https://github.com/ElektraInitiative/go-elektra) bindings _(Markus Raab)_
+- Fix order of tutorials _(Markus Raab)_
+- Added API-Reviews for multiple functions in the public API _(Stefan Hanreich)_
+- Minor rewording in [java-kdb.md](/doc/tutorials/java-kdb.md) _(@aaronabebe)_
+- Added a short Visual Studio 2019 tutorial (/doc/tutorials/contributing-windows.md) _(Dominic Jäger)_
+- Added hint regarding WSL filesystem configuration (/doc/tutorials/contributing-windows.md) _(@tucek)_
+- Fixed broken link in yanlr-plugin readme _(@lawli3t)_
```
Comment (mpranj) [inline doc/news/_preparation_next_release.md:165]: It seems something went wrong here with the rebase. Lots of changes are not from this PR (`doc/news/2021-04-12_0.9.5.md`).

Comment (robaerd) [reply doc/news/_preparation_next_release.md]: Thank you for pointing this out! Should now be fixed.

**Review by mpranj — APPROVED** [review]
Thank you, looks good to me!

I think we can merge it. @markus2330 can also review & request changes post-merge.

**Review by markus2330 — COMMENTED** [review]
Good job, it makes Elektra much easier for beginners :100:

**Pull request merged**